<launch>
    <arg name="namespace"               default="dynamixel_controller_1"/>
    <arg name="usb_port"                default="/dev/for_docker/screw"/>
    <arg name="dxl_baud_rate"           default="1000000"/>

    <arg name="use_moveit"              default="false"/>
    <arg name="use_joint_state"         default="true"/>
    <arg name="use_cmd_vel"             default="false"/>

    <arg name="num_controllers"       default="1"/>
    <arg name="config_file"            default="$(find o2ac_fastening_tools)/config/fastening_tools.yaml"/>

    <param name="dynamixel_info"          value="$(arg config_file)"/>

    <node name="$(arg namespace)" pkg="dynamixel_workbench_controllers" type="dynamixel_workbench_controllers"
         output="screen" args="$(arg usb_port) $(arg dxl_baud_rate)">
        <param name="use_moveit"              value="$(arg use_moveit)"/>
        <param name="use_joint_states_topic"  value="$(arg use_joint_state)"/>
        <param name="use_cmd_vel_topic"       value="$(arg use_cmd_vel)"/>
        <rosparam>
          publish_period: 0.010
          dxl_read_period: 0.010
          dxl_write_period: 0.010
          mobile_robot_config:                <!--this values will be set when 'use_cmd_vel' is true-->
            seperation_between_wheels: 0.160  <!--default value is set by reference of TB3-->
            radius_of_wheel: 0.033            <!--default value is set by reference of TB3-->
        </rosparam>
    </node>

    <node name="fastening_tool_action_server" pkg="o2ac_fastening_tools" type="fastening_tool_action_server.py" output="screen">
        <param name="tools_info"      type="string" value="$(arg config_file)"/>
        <param name="controller_ns"              value="$(arg namespace)"/>
    </node>
    
    <arg name="config_dir"            default="$(find o2ac_fastening_tools)/config"/>
    <!-- TODO: These arguments are kind of nonsense and could be cleaned up -->
    <arg name="suction_control"       default="suction_control"/>
    <arg name="operation_mode"        default="operation_mode"/>
    <node name="suction_controller" pkg="o2ac_fastening_tools" type="suction_controller.py" output="screen">
        <param name="config_dir"    type="string" value="$(arg config_dir)"/>
        <param name="suction_control" type="string" value="$(arg suction_control)"/>
        <param name="operation_mode" type="string" value="$(arg operation_mode)"/>
    </node>
</launch>
